### ISSUE TITLE
Get new GC in LuaJIT upstream

### ISSUE THREAD

**Issue by DemiMarie** [fsfod/LuaJIT#3]
It would be very useful to get the new GC upstreamed.

**Comment by fsfod** [comment]
It still missing lots of things and I got no reply from Mike Paul when I emailed him months ago

**Comment by lukego** [comment]
Seems like we need to create a new upstream branch, right? Mike is maintaining `master` and `v2.1` but these are both in feature-freeze and so somebody needs to maintain a new branch that accepts new features (`v2.2`, `v3.0`, etc).

I would be quite willing to do the branch-maintaining part of the work. For example I could maintain a branch that adopts a new GC once both @fsfod and @corsix are happy with the implementation and when it passes all available tests (I can use CI to verify LuaJIT suite, LuaJIT benchmarks, Snabb benchmarks, anything else that is available.)

This would only work on the assumption that the GC implementation does not require input from Mike and that @fsfod and @corsix see eye to eye on the implementation...?

**Comment by DemiMarie** [comment]
@lukego I agree.  `v2.1` did get an ARM64 JIT backend and fixed trace stitching, but that seems to be it.

Mike Pall has been looking for a maintainer for over a year.  Are you stepping up to the plate?
